src/tools/log_exporter.cc

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tools/log_exporter.h"
+
+#include <dirent.h>
+#include <gflags/gflags.h>
+#include <sched.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+
+#include "base/file_util.h"
+#include "base/glog_wapper.h"
+#include "log/log_reader.h"
+#include "log/log_writer.h"
+#include "proto/common.pb.h"
+#include "proto/tablet.pb.h"
+#include "storage/snapshot.h"
+#include "tools/tablemeta_reader.h"
+
+DEFINE_string(delimiter, ",", "delimiter");
+
+using ::openmldb::base::ParseFileNameFromPath;
+using ::openmldb::base::Slice;
+using ::openmldb::log::NewSeqFile;
+using ::openmldb::log::Reader;
+using ::openmldb::log::SequentialFile;
+using ::openmldb::log::Status;
+
+namespace openmldb {
+namespace tools {
+
+void Exporter::ReadManifest() {
+    std::string manifest_path = table_dir_path_ + "/snapshot/MANIFEST";
```

Comment:
check whether manifest_path is exist.  if `makesnapshot` has not executed, it will not generate `MANIFEST`

Reply:
On line 54, the GetLocalManifest function checks if manifest_path exists and returns 1 if not.